conan/cli/commands/cache.py

```diff
@@ -46,6 +48,8 @@ def cache_path(conan_api: ConanAPI, parser, subparser, *args):
             path = conan_api.cache.recipe_metadata_path(ref)
         else:
             raise ConanException(f"'--folder {args.folder}' requires a valid package reference")
+        if args.folder and not os.path.exists(path):
+            raise ConanException(f"'{args.folder}' folder does not exist for the reference {ref}")
```

Comment:
This check seems to be something particular about this command, should the API ensure that the folder exists? Not sure either.

> (side note, it seems message in the above line 50 is wrong)

No, it's not wrong but confusing. That's why I made that refactor.

Reply:
> This check seems to be something particular about this command, should the API ensure that the folder exists? Not sure either.

The problem could be the API returning the ``source`` folder path if it doesn't exist. What could be the use case? If the user writes something to that folder, that is undefined behavior, and Conan will not execute the ``source()`` method when necessary, because it thinks the sources are already (I know users shouldn't write in the cache in any case, but it is more likely to happen if the API gives them a folder path that they can write)

> No, it's not wrong but confusing. That's why I made that refactor.

Yes, make sense to try to improve it later in another PR (not urgent)